test/http-client.spec.js

```diff
@@ -534,60 +571,188 @@ describe('http client', () => {
+          this.called = true;
+          return message;
+        }
+      }
+
+      var interceptor = new RequestInterceptor();
+
+      var client = new HttpClient()
+        .configure(x => x.withBaseUrl(baseUrl))
+        .addInterceptor(interceptor);
+
+      client.get('some/cool/path')
+        .then((response) => {
+          expect(response.isSuccess).toBe(true);
+          expect(interceptor.called).toBe(true);
+          done();
+        })
+    });
+
+    it('should intercept response message', (done) => {
+      class ResponseInterceptor {
+        constructor() {
+          this.called = false;
+        }
+
+        response(message) {
+          this.called = true;
+          return message;
+        }
+      }
+
+      var interceptor = new ResponseInterceptor();
+
+      var client = new HttpClient()
+        .configure(x => x.withBaseUrl(baseUrl))
+        .addInterceptor(interceptor);
+
+      client.get('some/cool/path')
+        .then((response) => {
+          expect(response.isSuccess).toBe(true);
+          expect(interceptor.called).toBe(true);
+          done();
+        })
+    });
+
+    it('should intercept request errors', (done) => {
+      class RequestErrorInvokerInterceptor {
+        request(message) {
+          throw new Error('error');
+        }
+      }
+
+      class RequestErrorInterceptor {
+
+        constructor() {
+          this.called = false;
+        }
+
+        requestError(error) {
+          this.called = true;
+          // Re-throw
+          throw error;
+        }
 
-      var response = jasmine.Ajax.requests.mostRecent();
-      expect(response.upload.onprogress).toBe(callback);
+      }
+
+      var interceptor = new RequestErrorInterceptor();
+
+      var client = new HttpClient()
+        .configure(x => x.withBaseUrl(baseUrl))
+        .addInterceptor(new RequestErrorInvokerInterceptor()) // Simulate requestError
+        .addInterceptor(interceptor);
+
+      client.get('some/cool/path')
+        .catch((error) => {
+          expect(error.message).toBe('error');
+          expect(interceptor.called).toBe(true);
+          done();
+        })
+    });
+
+    it('should intercept response errors', (done) => {
+      jasmine.Ajax
+        .stubRequest('http://example.com/some/cool/path')
+        .andReturn({ status: 500 });
+
+      class ResponseErrorInterceptor {
+
+        constructor() {
+          this.called = false;
+        }
+
+        responseError(response) {
+          this.called = true;
+          // Re-throw
+          throw response;
+        }
+
+      }
+
+      var interceptor = new ResponseErrorInterceptor();
+
+      var client = new HttpClient()
+        .configure(x => x.withBaseUrl(baseUrl))
+        .addInterceptor(interceptor);
+
+      client.get('some/cool/path')
+        .catch((response) => {
+          expect(response.isSuccess).toBe(false);
+          expect(interceptor.called).toBe(true);
+          done();
+        })
     });
```

Comment:
I think it'd be great to have a test that verifies execution order of interceptors when you have multiple registered.